### PR TITLE
Add tests infrastructure

### PR DESCRIPTION
**Pull request by neithernut**
This PR adds very basic infrastucture which will serve as a prequisite for a variety of tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,6 +33,9 @@ Makefile
 *.cmake
 *.stamp
 
+# Test related stuff
+Testing
```
Comment (matthiasbeyer) [inline .gitignore:38]: Nope. The directory for tests will be called `tests`.

File: .travis.yml [hunk 2]
```diff
@@ -21,3 +21,4 @@ script:
     - cmake -DHARD_MODE=ON .
     - make
     - make paper
+    - ctest -VV .
```
Comment (matthiasbeyer) [inline .travis.yml:25]: What does this?

**Comment by matthiasbeyer** [comment]
Thanks for this PR. I waited for this!

---

Can you add a program to the test setup? We need "check" (archlinux package name) for testing.

File: .gitignore (hunk 1 above)
Comment (matthiasbeyer) [inline .gitignore:38]: Or does cmake store things in here?

**Comment by neithernut** [comment]
See PR #58.

File: .travis.yml (hunk 2 above)
Comment (neithernut) [inline .travis.yml:25]: It will run the tests and print all the output of the testing programs
...once they exist.

**Comment by neithernut** [comment]
I'll add build and execution of the dummy test to this PR, so hang on...

**Comment by matthiasbeyer** [comment]
:+1:

**Comment by neithernut** [comment]
I'll notify you here when the work is done.

**Comment by neithernut** [comment]
If the CI gives us a green light, the branch may now be merged.
I'd like to reorder a few commits before the merge, though.

**Comment by neithernut** [comment]
This PR is related to issue #16.

**Comment by neithernut** [comment]
Since you're waiting for this, I took the liberty of reordering, rewording and squashing some commits.
The branch is now ready for merge.

**Comment by matthiasbeyer** [comment]
Looks good to me, thanks!

**Comment by matthiasbeyer** [comment]
Narg. You link to the library.so? So there is no way I can tests the linkedlist for example?

Or how to achieve this? Can you please add some patches on #58 ?

**Comment by neithernut** [comment]
Mh... the ll-symbols may indeed not be exported...
I'll add a fix.

**Comment by matthiasbeyer** [comment]
I meant the include path for libreset is not exported so I can not do `#include "libreset/ll.h"` in my test.

**Comment by neithernut** [comment]
Yes, .I'll fix that one, too

**Comment by neithernut** [comment]
Having the test program link to all the object files from the library or a test-library, which exports all symbols, may a bit more tricky than anticipated --at least for a _nice_ solution.
I'll push a fix for the include directory-issue only.

**Comment by neithernut** [comment]
See PR #93.

**Pull request merged**